### DISCUSSION TITLE
Splitting oasdiff into 2 repos

### DISCUSSION THREAD

**Discussion by reuvenharrison** [Tufin/oasdiff#319] [Ideas]
Hi all,
We recently released v2 of oasdiff.
The motivation behind this was to notify our users about a breaking change in the new v2 cmd-line with sub commands (https://github.com/Tufin/oasdiff/discussions/292).
However, what we didn't realize is, as a go package, releasing v2 must follow certain guidelines: https://go.dev/blog/v2-go-modules
As a result, users of the oasdiff golang package cannot consume v2 (see: https://github.com/Tufin/oasdiff/issues/318).
Apart from fixing this, I would also like to suggest that we separate the go package from the cmd-line.
The current repo will remain the home of the oasdiff cmd-line and a new repo will host the oasdiff go package.
This will enable us to focus on the different considerations of each audience moving forward, and, hopefully, avoid similar problems.
Your feedback is welcome.
Reuven

**Comment by wtrocki** [comment]
I been in situations couple times I would recomend to consider using golang modules (aka go monorepo). There would be little benefits of moving to the separate repo when cli can be an submodule with different release tags and independent versioning. 
This specifically applies to the CLI where version doesn't matter much as people consume artifacts instead.
Separate repository has tons of extra boilerplate work and sync so it might also affect the wider community.

**Reply by reuvenharrison** [reply]
After discussing and learning more about the implications of golang v2, I am going to revert this and go back to v1. Thanks all.
